app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountMongoDetailsAction.scala

```diff
@@ -35,26 +37,46 @@ case class RequestWithUserDetailsFromSessionAndMongo[A](request: Request[A],
                                                         accountDetailsFromMongo: AccountDetailsFromMongo)
   extends WrappedRequest[A](request)
 
+object RequestWithUserDetailsFromSessionAndMongo{
+  import scala.language.implicitConversions
+  implicit def requestConversion(
+                                  requestWithUserDetailsFromSessionAndMongo: RequestWithUserDetailsFromSessionAndMongo[_])
+  : RequestWithUserDetailsFromSession[_] = {
+    RequestWithUserDetailsFromSession(
+      requestWithUserDetailsFromSessionAndMongo.request,
+      requestWithUserDetailsFromSessionAndMongo.userDetails,
+      requestWithUserDetailsFromSessionAndMongo.sessionID)
+  }
+}
+
 trait AccountMongoDetailsActionTrait
     extends ActionRefiner[RequestWithUserDetailsFromSession, RequestWithUserDetailsFromSessionAndMongo]
 
-class AccountMongoDetailsAction @Inject()(accountCheckOrchestrator: AccountCheckOrchestrator,
+
+  class AccountMongoDetailsAction @Inject()(accountCheckOrchestrator: AccountCheckOrchestrator,
                                           val parser: BodyParsers.Default,
-                                          errorHandler: ErrorHandler)(implicit val executionContext: ExecutionContext) extends AccountMongoDetailsActionTrait {
+                                          errorHandler: ErrorHandler,
+                                            appConfig: AppConfig)(implicit val executionContext: ExecutionContext) extends AccountMongoDetailsActionTrait {
   implicit val baseLogger: Logger = Logger(this.getClass.getName)
 
   override protected def refine[A](request: RequestWithUserDetailsFromSession[A]): Future[Either[Result, RequestWithUserDetailsFromSessionAndMongo[A]]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
-    accountCheckOrchestrator.getAccountType(executionContext, hc, request).value
-      .map {
-        case Right(accountType) => Right(
+    (for {
+      redirectUrl <- accountCheckOrchestrator.getRedirectUrlFromCache(request)
+      accountType <- accountCheckOrchestrator.getAccountTypeFromCache(request, implicitly)
+    } yield {
+      val resultOfGettingCache = (redirectUrl -> accountType)
+      resultOfGettingCache match {
+        case (Some(redirectUrl), Some(accountType)) => Right(
           RequestWithUserDetailsFromSessionAndMongo(
             request.request, request.userDetails, request.sessionID, AccountDetailsFromMongo(accountType)
           )
         )
-        case Left(_) =>
-          Left(errorHandler
-            .handleErrors(UnexpectedErrorWhenGettingUserType, "[AccountTypeAction][invokeBlock]")(request, baseLogger))
+        case _ => Left(Redirect(routes.AccountCheckController.accountCheck(appConfig.redirectPTAUrl)))
```

Comment:
This should only be redirecting to accountCheck if the redirectUrl is in the cache. Otherwise we should render the error page. We shouldn't be hard coding the PTA url because in the future they may not come from PTA. We should be returning to where they ask us to return to.